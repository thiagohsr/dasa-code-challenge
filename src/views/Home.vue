<template>
  <div v-bind:class="classes.home">
    <h1>Busca de repositórios por usuário</h1>
    <p>
      Informe o nome do usuário que deseja visualizar os repositórios abaixo:
    </p>
    <github-search />

    <list-pagination />
    <h3>Exibindo repositórios para o usuário: {{ githubUser }}</h3>
    <list-repositories />

    <p v-if="errorMessage">{{ errorMessage }}</p>

    <list-pagination />
  </div>
</template>

<script>
import store from "@/store";
import classes from "@/utils/cssTranspilation";
import ListPagination from "@/components/ListPagination.vue";
import ListRepositories from "@/components/ListRepositories.vue";
import GithubSearch from "@/components/GithubSearch.vue";

const styles = {
  home: {
    display: "flex",
    flexDirection: "column",
    paddingLeft: 15,
    paddingTop: 15,
    paddingRight: 15,
    paddingBottom: 15
  },
  searchBox: {
    display: "inline-flex",
    justifyContent: "center"
  }
};

export default {
  name: "home",
  computed: {
    githubUser: () => store.state.githubUser,
    errorMessage: () => store.state.errorMessage
  },
  data() {
    return {
      classes: classes(styles)
    };
  },
  components: {
    "github-search": GithubSearch,
    "list-pagination": ListPagination,
    "list-repositories": ListRepositories
  }
};
</script>
