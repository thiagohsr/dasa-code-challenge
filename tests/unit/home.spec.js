import { shallowMount, mount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import Home from "@/views/Home.vue";

const localVue = createLocalVue();

localVue.use(Vuex);

describe("Home.vue", () => {
  let store;
  let state;
  beforeEach(() => {
    state = {
      githubUser: "",
      errorMessage: "",
      userRepositories: [],
      paginationLinks: {},
      isLoading: false
    };
    store = new Vuex.Store({
      state
    });
  });

  it("should render properly Home view", () => {
    const wrapper = shallowMount(Home, { store, localVue });
    const html = wrapper.vm.$el;
    expect(wrapper.name()).toBe("Home");
    expect(html).toMatchSnapshot();
  });

  it("should Home view render properly child components", () => {
    const wrapper = shallowMount(Home, { store, localVue });
    const searchName = wrapper.find("github-search-stub").name();
    const paginationName = wrapper.find("list-pagination-stub").name();
    const repositoriesName = wrapper.find("list-repositories-stub").name();
    expect(searchName).toBe("GithubSearch");
    expect(paginationName).toBe("ListPagination");
    expect(repositoriesName).toBe("ListRepositories");
  });

  it("should computed githubUser properly render", () => {
    const expectedValue = "thiagohsr";
    const wrapper = mount(Home, {
      store,
      localVue,
      computed: {
        githubUser() {
          return expectedValue;
        },
        errorMessage() {
          return "";
        }
      }
    });
    const userElement = wrapper.find(".githubUser");
    expect(userElement.text()).toContain(expectedValue);
  });

  it("should computed errorMessage properly render", () => {
    const expectedValue = "Mensagem de erro";
    const wrapper = mount(Home, {
      store,
      localVue,
      computed: {
        errorMessage() {
          return expectedValue;
        }
      }
    });
    const errorElement = wrapper.find(".errorMessage");
    expect(errorElement.text()).toContain(expectedValue);
  });
});
