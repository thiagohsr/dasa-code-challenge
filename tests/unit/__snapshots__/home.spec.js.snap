// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue should render properly Home view 1`] = `
<div
  class="home-0-0-1"
>
  <h1>
    Busca de reposit칩rios por usu치rio
  </h1>
   
  <p>
    
    Informe o nome do usu치rio que deseja visualizar os reposit칩rios abaixo:
  
  </p>
   
  <github-search-stub />
   
  <list-pagination-stub />
   
  <!---->
   
  <list-repositories-stub />
   
  <!---->
   
  <list-pagination-stub />
</div>
`;
